mlx-rs/src/ops/other.rs

```diff
@@ -91,13 +92,50 @@ pub fn diagonal_device(
     a.diagonal_device(offset, axis1, axis2, stream)
 }
 
+/// Perform the Einstein summation convention on the operands.
+///
+/// # Params
+///
+/// - subscripts: Einstein summation convention equation
+/// - operands: input arrays
+/// - stream: stream or device to evaluate on
+#[default_device]
+pub fn einsum_device(
+    subscripts: &str,
+    operands: &[&Array],
```

Comment:
We could probably make this something like `impl IntoIterator<Item = &Array>` since all it needs right now is iter and collect